tests/e2e/framework/kubernetes.go

```diff
@@ -0,0 +1,230 @@
+		NamespaceCreated: false,
+		TmpDir:           tmpDir,
+		YamlDir:          tmpDir + "/yaml",
+	}
+}
+
+func (k *KubeInfo) Setup() error {
+	if err := util.CreateNamespace(k.Namespace); err != nil {
+		glog.Error("Failed to create namespace.")
+		return err
+	}
+	k.NamespaceCreated = true
+	if err := k.deployIstio(); err != nil {
+		glog.Error("Failed to deployIstio.")
+		return err
+	}
+
+	glog.Info("Kubernetes setup finished.")
+	return nil
+}
+
+func (k *KubeInfo) Teardown() error {
+	if k.NamespaceCreated {
+		if err := util.DeleteNamespace(k.Namespace); err != nil {
+			return err
+		}
+		k.NamespaceCreated = false
+		glog.Infof("Namespace %s deleted", k.Namespace)
+	}
+	glog.Info("Uploading log remotely")
+	glog.Flush()
+	return nil
+}
+
+// Deploy istio modules
+func (k *KubeInfo) deployIstio() error {
+	if err := os.Mkdir(k.YamlDir, os.ModeDir|os.ModePerm); err != nil {
+		return err
+	}
+	usr, eUser := user.Current()
+	if eUser != nil {
+		return eUser
+	}
+	homeDir := usr.HomeDir
+
+	// Download Istioctl binary
+	util.WebDownload(k.TmpDir+"/istioctl", k.IstioCtlUrl+"/istioctl-linux")
+	if err := os.Chmod(fmt.Sprintf("%s/istioctl", k.TmpDir), 0755); err != nil {
+		return err
+	}
+	k.IstioCtl = fmt.Sprintf("%s/istioctl -c %s/.kube/config", k.TmpDir, homeDir)
+	glog.Infof("Downloaded istioctl to %s/istioctl\n", k.TmpDir)
+
+	if err := k.deployCore("manager.yaml"); err != nil {
+		return err
+	}
+	if err := k.deployCore("mixer.yaml"); err != nil {
+		return err
+	}
+	if err := k.deployCore("ingress-proxy.yaml"); err != nil {
+		return err
+	}
+	if err := k.deployCore("egress-proxy.yaml"); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Deploy istio module from yaml files
+func (k *KubeInfo) deployCore(name string) error {
+	yamlFile := k.TmpDir + "/yaml/" + name
+	if err := util.Fill(yamlFile, name+".tmpl", *k); err != nil {
+		glog.Errorf("Failed to fill %s", yamlFile)
+		return err
+	}
+	if err := util.KubeApply(k.Namespace, yamlFile); err != nil {
+		glog.Errorf("Kubectl apply %s failed", yamlFile)
+		return err
+	}
+
+	return nil
+}
+
+// Deploy testing app
+func DeployApp(k *KubeInfo, deployment, svcName, port1, port2, port3, port4, version string, injectProxy bool) error {
+	yamlFile := k.TmpDir + "/yaml/" + svcName + "-app.yaml"
+	if err := util.Fill(yamlFile, "app.yaml.tmpl", map[string]string{
+		"Hub":        k.Hub,
+		"Tag":        k.Tag,
+		"service":    svcName,
+		"deployment": deployment,
+		"port1":      port1,
+		"port2":      port2,
+		"port3":      port3,
+		"port4":      port4,
+		"version":    version,
+	}); err != nil {
+		glog.Errorf("Failed to generate yaml for service %s in deployment %s", svcName, deployment)
+		return err
+	}
+
+	if injectProxy {
+		injectedYamlFile := k.TmpDir + "/yaml/injected-" + svcName + "-app.yaml"
+		if _, err := util.Shell(fmt.Sprintf("%s kube-inject -f %s -o %s --hub %s --tag %s -n %s", k.IstioCtl, yamlFile, injectedYamlFile, k.ProxyHub, k.ProxyTag, k.Namespace)); err != nil {
```

Comment:
Might be useful to create a istioctl library that takes care of downloading the istioctl and etc..